### PR TITLE
Add `pre-commit` hook to keep dictionary file sorted

### PR DESCRIPTION
**Pull request by alexdewar**
I've added a pre-commit hook to keep our dictionary file sorted, which should make it a bit tidier. It will also filter out duplicate entries.

I wanted it to be sorted in a case-insensitive manner, but this didn't work for me: https://github.com/pre-commit/pre-commit-hooks/issues/1105

### CODE REVIEW

File: .wordlist.txt [hunk 1]
```diff
@@ -1,139 +1,118 @@
...
 Manani
 Matejuk
-math
 Mathjax
-mdx
 Mirai
-mirai
 MkDocs
-MLOps
+Modeling
```
Comment (dc2917) [inline .wordlist.txt:96]: 🤢

File: .wordlist.txt [hunk 2]
```diff
@@ -144,131 +123,152 @@ RSLondonSouthEast
...
-Yee
-GitHub
-Crypto
-CrystalBot
-CryptoTrade
-Gueant
-LinkedIn
+jamesturner
+jcohen
+labeled
```
Comment (dc2917) [inline .wordlist.txt:223]: 🤢

**Review by dc2917 — APPROVED** [review]
Nice!

**Review by HarmonicReflux — APPROVED** [review]
LGTM to get the job done.

`.wordlist.txt` is somewhat hacky and get worse as we go, but working with spellchecker has been frustrating from my side: It checks what it should not and it is difficult to modify.

**Review by dalonsoa — APPROVED** [review]
This is handy 👍 

`Álvarez` (my second surname) is the last on the list 😢

**Review by jamesturner246 — APPROVED** [review]
Cool. LGTM. :+1:

**Comment by alexdewar** [comment]
> `Álvarez` (my second surname) is the last on the list 😢

Yeah. Clearly the sorting isn't Unicode aware :disappointed:

**Comment by HarmonicReflux** [comment]
> This is handy 👍
> 
> `Álvarez` (my second surname) is the last on the list 😢

Well, thinking about it @dalonsoa, this will ensure that everyone checking the file reads your name (as hardly anyone ever scrolls up to the beginning of it).

**Review by Scutua — APPROVED** [review]

**Pull request merged**